src/mon/ConfigKeyService.cc

```diff
@@ -218,6 +218,69 @@ bool ConfigKeyService::service_dispatch(MonOpRequestRef op)
     // return for now; we'll put the message once it's done.
     return true;
 
+  } else if (prefix == "config-key mput" ||
+	     prefix == "config-key mput-prefix") {
+    if (!mon->is_leader()) {
+      mon->forward_request_leader(op);
+      // we forward the message; so return now.
+      return true;
+    }
+
+    string key_prefix;
+    if (prefix == "config-key mput-prefix") {
+      if (!cmd_getval(g_ceph_context, cmdmap, "key_prefix", key_prefix)) {
+	ret = -EINVAL;
+	goto out;
+      }
+      if (key_prefix.size() == 0) {
+	ss << "key_prefix must be non-empty";
+	ret = -EINVAL;
+	goto out;
+      }
+    }
+    vector<string> kv;
+    if (!cmd_getval(g_ceph_context, cmdmap, "keysandvalues", kv)) {
+      ret = -EINVAL;
+      goto out;
+    }
+    if (kv.empty() || kv.size() % 2) {
```

Comment:
I think it would be nicer if these k/v pairs were separated by '='. Easier to read, and imo less prone to error if many pairs are being manually written in the cli. And I think it would still just be a matter of using the vector, and run each element through `get_str_map()` until we run out of elements in the vector.